Incorporating vote history into transformers

pjsier opened the pull request:
# What's in this PR?

This one definitely needs a lot of review. Trying to include vote history as part of transformers while keeping behavior relatively separate.

* Adds kwarg `history` to multiple methods to flag different behavior.
* Adds `history_type_dict` and `history_fields` which are referenced in methods when `history` is flagged.
* Evaluate `hist_`-prefixed methods similar to how `extract` methods are currently used.
* Update `CsvOutput` to be able to run voter history with `--history` flag.
* Implementations for Ohio and Michigan of reading voter history files with these changes. 
* Removed old scripts for transforming vote history.

Overall, it reuses most of the behavior to keep things as consistent as possible, with the main change that will change across states being the generator returned to run vote history which doesn't seem like we'll be able to entirely generalize.

## References

Progress on: #97

slavingia commented:
merge conflicts to address btw

slavingia commented on src/python/national_voter_file/us_states/mi/transformer.py:
```diff
@@ -79,11 +94,21 @@ def __init__(self, input_path, *args):
-            return self.zip_voters(
-                os.path.join(os.path.dirname(self.input_path), 'entire_state_v.lst')
-            )
+            if not self.history:
```
reverse this so the positive condition is first? a little bit easier to understand that way i think.

slavingia commented:
this will be great fodder for newcomers like the transformers themselves are/were. we can take all the states we have voting history for and 💥 a bunch of easy tasks to understand the loading stuff while we clean up the other bits.

pjsier commented:
Great! I'll add some specs and tweak that if statement (I put it first because it's going to be less common for it to be flagged, but you're right that it's harder to read). 

The only question I have is on using the `history=False` flag on methods rather than making `history` an attribute of a `StateTransformer`. I made it an argument to the methods because at least Ohio, Utah, and Delaware have vote history items in the same file as voter information (as separate columns), so I think it makes more sense to have it as a method and just re-run the transformer rather create a new transformer object to switch. Wanted to flag it in case anyone disagrees though.

pjsier commented:
Added tests for vote history transformers and set them up for Michigan and Ohio. Also just removed `history_fields`, because if they have columns with election names that will change we really can't provide a set `fieldnames` argument anyway. 

The history test is pretty general for now, and mostly like the normal transformer test, but I wanted to separate it because the setup may be a lot different for other states.
